code/modules/fish/fishtank.dm

```diff
@@ -292,37 +283,26 @@
 	if(cur_health <= 0)								//The tank is broken, destroy it
 		destroy()
 
-/obj/machinery/fishtank/proc/kill_fish(var/type = null)
+/obj/machinery/fishtank/proc/kill_fish(datum/fish/fish_type = null)
 	//Check if we were passed a fish to kill, otherwise kill a random one
-	if(type)
-		fish_list.Remove(type)						//Kill a fish of the specified type
-		fish_count --								//Lower fish_count to reflect the death of a fish, so the everything else works fine
-	else
-		fish_list.Remove(pick(fish_list))			//Kill a random fish
-		fish_count --								//Lower fish_count to reflect the death of a fish, so the everything else works fine
-
-/obj/machinery/fishtank/proc/add_fish(var/type = null)
+	if(!fish_type)
+		fish_type = pick(fish_list)
+	fish_list.Remove(fish_type)						//Kill a fish of the specified type
+	fish_count --									//Lower fish_count to reflect the death of a fish, so the everything else works fine
+	if(istype(type, /datum/fish/glofish))
+		adjust_light()
+
+/obj/machinery/fishtank/proc/add_fish(datum/fish/fish_type = null)
 	//Check if we were passed a fish type
-	if(type)
-		fish_list.Add("[type]")						//Add a fish of the specified type
+	if(fish_type)
+		fish_list.Add(fish_type)					//Add a fish of the specified type
 		fish_count++								//Increase fish_count to reflect the introduction of a fish, so the everything else works fine
 		//Announce the new fish
-		src.visible_message("A new [type] has hatched in \the [src]!")
+		src.visible_message("A new [fish_type.fish_name] has hatched in \the [src]!")
```

Comment:
I'm leaving this as-is because I hate the look of `[]` in strings, this passed review when it was added, and is a generally-used way of doing things throughout the code.